includes/compatibility/class-wcpdf-compatibility-third-party-plugins.php

```diff
@@ -79,6 +74,30 @@ public function reset_invoice_data ( $order ) {
 		WCX_Order::delete_meta_data( $order, '_wcpdf_invoice_exists' );
 	}
 
+	/**
+	 * Removes documents meta from WooCommerce Subscriptions renewal order
+	 */
+	public function wcs_renewal_order_meta ( $meta, $to_order, $from_order ) {
+		if ( ! empty( $meta ) ) {
+			$documents = WPO_WCPDF()->documents->get_documents();
+			foreach( $documents as $document ) {
+				foreach( $meta as $key => $value ) {
+					$document_meta = array(
+						"_wcpdf_{$document->slug}_number",
+						"_wcpdf_{$document->slug}_number_data",
+						"_wcpdf_formatted_{$document->slug}_number",
+						"_wcpdf_{$document->slug}_date",
+						"_wcpdf_{$document->slug}_exists",
+					);
```

Comment:
There's others too, see `$document->delete()`:
https://github.com/wpovernight/woocommerce-pdf-invoices-packing-slips/blob/ec7b056ff3a4344f85ab641f477663bcade56848/includes/documents/abstract-wcpdf-order-document.php#L292-L302

And since the renewal could be based on very old legacy orders it should also still include the `"_wcpdf_formatted_{$document->slug}_number"` we had before.

Using `array_map` we could reuse the same `wpo_wcpdf_delete_document_data_keys` filter and keep the rest of your logic the same (not tested):
```php
$document_data_keys = apply_filters( 'wpo_wcpdf_delete_document_data_keys', array( 
	'settings', 
	'date', 
	'date_formatted', 
	'number', 
	'number_data', 
	'notes', 
), $document );

$document_meta = array_map( function( $data_key ) use ( $document ) {
	return "_wcpdf_{$document->slug}_{$data_key}";
}, $document_data_keys );
$document_meta[] = "_wcpdf_formatted_{$document->slug}_number"; // legacy meta key
```